fortex/health/processors/ner_label_processor.py

```diff
@@ -0,0 +1,148 @@
+    "en_core_sci_md": "https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy"
+    "/releases/v0.3.0/en_core_sci_md-0.3.0.tar.gz",
+    "en_core_sci_lg": "https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy"
+    "/releases/v0.3.0/en_core_sci_lg-0.3.0.tar.gz",
+    "en_ner_craft_md": "https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy"
+    "/releases/v0.3.0/en_ner_craft_md-0.3.0.tar.gz",
+    "en_ner_jnlpba_md": "https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy"
+    "/releases/v0.3.0/en_ner_jnlpba_md-0.3.0.tar.gz",
+    "en_ner_bc5cdr_md": "https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy"
+    "/releases/v0.3.0/en_ner_bc5cdr_md-0.3.0.tar.gz",
+    "en_ner_bionlp13cg_md": "https://s3-us-west-2.amazonaws.com/ai2-s2"
+    "-scispacy/releases/v0.3.0/en_ner_bionlp13cg_md-0"
+    ".3.0.tar.gz",
+}
+
+
+def load_lang_model(lang_model):
+    # pylint: disable=import-outside-toplevel
+    if lang_model in CUSTOM_SPACYMODEL_URL:
+        # download ScispaCy model using URL
+        import subprocess
+        import sys
+        import os
+        import importlib
+
+        download_url = CUSTOM_SPACYMODEL_URL[lang_model]
+        command = [sys.executable, "-m", "pip", "install"] + [download_url]
+        subprocess.run(
+            command, env=os.environ.copy(), encoding="utf8", check=False
+        )
+        cls = importlib.import_module(lang_model)
+        return cls.load()
+    else:
+        # Use spaCy download
+        try:
+            nlp = spacy.load(lang_model)
+        except OSError:
+            download(lang_model)
+            nlp = spacy.load(lang_model)
+    return nlp
+
+
+class NERLabelProcessor(PackProcessor):
+    r"""
+    Implementation of this NERLabelProcessor has been based on spaCy
+    pretained model. A rendition of it that exists on github has
+    been referred to as well.
+
+    Referred repository link:
+    https://github.com/explosion/spaCy
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.nlp = None
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.nlp = load_lang_model("en_ner_bc5cdr_md")
+
+    def _process(self, input_pack: DataPack):
+        r"""
+        NER Label processing is based on spaCy.
+        """
+        labels = self.configs.labels
+
+        doc = input_pack.text
+
+        # Do all process.
+        if self.nlp is None:
+            raise ProcessExecutionException(
+                "The SpaCy pipeline is not initialized, maybe you "
+                "haven't called the initialization function."
+            )
+        result = self.nlp(doc)
+
+        if "disease" in labels:
+            for ent in result.ents:
+                if ent.label_ == "DISEASE":
+                    Disease(
+                        pack=input_pack, begin=ent.start_char, end=ent.end_char
+                    )
+
+        if "chemical" in labels:
+            for ent in result.ents:
```

Comment:
We need not iterate through the entities twice right? Or n number of times for n different annotations. We can just iterate once and check for each entity if their label exists in the labels that we are working with (Disease and Chemical for now).